.circleci/config.yml

```diff
@@ -149,25 +146,29 @@ jobs:
     steps:
       - checkout
       - restore_module_cache
-      - persist_to_workspace:
-          root: ~/
-          paths: project
+      - run:
+          name: Install deps
+          command: make for-all CMD='go mod download'
       - run:
           name: Install tools
           command: make install-tools
       - run:
           name: Install testbed tools
           command: make -C testbed install-tools
+      - save_module_cache
       - persist_to_workspace:
           root: ~/
-          paths: go/bin
+          paths:
+            - project
+            - go/bin
+            - go/pkg/mod
   lint:
     executor: golang
     steps:
       - attach_to_workspace
       - run:
           name: Lint
-          command: CMD="make lint" make -j4 for-all
+          command: CMD="make lint" make -j16 for-all
```

Comment:
This is another thing I noticed. I don't think -j will increase parallelism here because the looping is being done in bash embedded in the make. I think we'd have to change how the make target works for the parallelism to work.

Reply:
```bash
for-all:
	@$${CMD}
	@set -e; for dir in $(ALL_MODULES); do \
	  (cd "$${dir}" && \
	  	echo "running $${CMD} in $${dir}" && \
	 	$${CMD} ); \
	done
```

There's no way for make to parallelize this afaict.

Reply:
It doesn't. Was working on a new target does actually runs everything in parallel. It's in now. I left `for-all` as is and named it `for-all-make` since it runs a make target.

Reply:
Got it to work but looks like `golangci-lint` doesn't like to run in parallel 🤦 

```
Static check finished successfully
 
golangci-lint run
 
ERRO Parallel golangci-lint is running            
 
../../Makefile.Common:95: recipe for target 'lint' failed
 
make[1]: *** [lint] Error 3
 
make[1]: Leaving directory '/home/circleci/project/exporter/honeycombexporter'
 
Makefile:67: recipe for target 'for-all-make-./exporter/honeycombexporter' failed
 
make: *** [for-all-make-./exporter/honeycombexporter] Error 2
 
make: *** Waiting for unfinished jobs....
 
ERRO Parallel golangci-lint is running            
 
ERRO Parallel golangci-lint is running            
 
../../Makefile.Common:95: recipe for target 'lint' failed
 
make[1]: *** [lint] Error 3
 
make[1]: Leaving directory '/home/circleci/project/exporter/splunkhecexporter'
 
ERRO Parallel golangci-lint is running 
```

Reply:
Upgraded golangci-lint. Fixed now.

Reply:
Reduced to 3 minutes from 5-7 minutes in master.

Reply:
Would `for-all-target` maybe be more clear what it's for and naming CMD in it TARGET? Just an idea.

Reply:
I thought of that but felt using same var was slightly nicer when browsing CLI history and just changing `for-all` to `for-all-make`. But it does indeed look better. Will change.

Reply:
One other option is to make for-all itself support parallelism using xargs:

```makefile
.PHONY: for-all
PARALLEL?=1
for-all:
	@echo $(ALL_MODULES) | xargs -I% -n1 -P$(PARALLEL) sh -ec "echo 'running $(CMD) in %' ; cd % && $(CMD)"
```

The native make make one will probably be faster (less forking) but I'd be surprised if it's by much. Having the for-all one support parallel is nice for cases where the CMD isn't a Makefile target. Both probably have their uses. Don't have to add it if you don't need it but I probably will at some point.

Reply:
Actually maybe the native one won't be faster since it's still doing an invocation for each subdirectory 🤔

Reply:
Don't see much value in changing it. People might be using it in the local dev flow so don't want to cause any annoyances. make based parallel build targets works as expected and integrates with the intuitive `-jX` flag as well so I think we should keep that separate.